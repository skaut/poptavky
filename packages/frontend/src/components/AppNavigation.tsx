import type React from "react";

import { Navigation } from "./Navigation";

export const AppNavigation = (): React.JSX.Element => (
  <Navigation
    items={[
      {
        link: "/",
        title: "PoptÃ¡vky",
      },
      {
        link: "/projekty",
        title: "Projekty",
      },
    ]}
  />
);
