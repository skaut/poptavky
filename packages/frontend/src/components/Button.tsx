import React from "react"
import styled from "@emotion/styled"
import { theme } from "../theme"
import { ExtLink, LinkProps } from "./Link"

interface Props extends LinkProps {}

export const Button: React.FC<Props> = (props) => <ButtonWrapper {...props} />

const ButtonWrapper = styled(ExtLink)`
  display: inline-block;
  font-size: 1.2rem;
  background-color: ${theme.colors.brand};
  color: white;
  margin-top: 0.2rem;
  padding: 0.6rem 1.2rem;
  border-radius: 10px;
  box-shadow: 0 0 0 ${theme.colors.gray};
  transition: box-shadow 300ms;
`
