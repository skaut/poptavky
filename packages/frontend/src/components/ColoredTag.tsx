import { css } from "@emotion/react";
import styled from "@emotion/styled";
import stringToColor from "string-to-color";

export const ColoredTag = styled("span")(
  (props: { children: string; isLight?: boolean }) => css`
    background-color: ${stringToColor(props.children)}33;
    color: ${props.isLight === true ? "white" : "#444;"};
    font-family: themix;
    font-size: 0.8rem;
    margin-right: 8px;
    padding: 0.2rem 0.4rem;
    border-radius: 4px;
    ${props.isLight === true ? "border: 1px solid #ffffff55" : ""};
  `,
);
