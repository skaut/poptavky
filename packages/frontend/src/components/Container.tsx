import { Global } from "@emotion/react";
import styled from "@emotion/styled";
import type React from "react";
import { Link, Outlet } from "react-router-dom";

import { globalStyles } from "../globalStyles";
import logo from "../images/logo.svg";
import { theme } from "../theme";
import { ScrollToTop } from "./ScrollToTop";

const WrapperDiv = styled("div")`
  max-width: ${theme.layout.width}px;
  padding: 0 16px;
  margin: 0 auto;
`;

const Header = styled("header")`
  display: flex;
  align-items: center;
  height: 60px;
  margin: 0 auto;
  padding: 16px 0;
`;

const Logo = styled("img")`
  max-height: 100%;
`;

const Title = styled("h1")`
  font-family: themix;
  font-size: 20px;
  margin: 0 0 0 24px;
`;

export const Container = (): React.JSX.Element => (
  <WrapperDiv>
    <Global styles={globalStyles} />
    <Link title="domů" to="/">
      <Header>
        <Logo alt="logo" src={logo} />
        <Title>IT poptávky</Title>
      </Header>
    </Link>
    <main>
      <ScrollToTop />
      <Outlet />
    </main>
  </WrapperDiv>
);
