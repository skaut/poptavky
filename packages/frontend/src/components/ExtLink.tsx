/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import type React from "react";

export interface LinkProps {
  readonly children: React.ReactNode;
  readonly href: string;
  readonly title?: string;
  readonly targetSelf?: boolean;
  readonly className?: string;
}
export const ExtLink = ({
  children,
  href,
  title,
  targetSelf,
  className,
}: LinkProps): React.JSX.Element => (
  <a
    className={className}
    href={href}
    title={title}
    {...(targetSelf !== true
      ? {
          target: "_blank",
          rel: "noreferrer noopener",
        }
      : {})}
    css={css`
      display: inline-flex;
      align-items: center;
    `}
  >
    {children}
  </a>
);

ExtLink.defaultProps = {
  className: undefined,
  targetSelf: false,
  title: undefined,
};
