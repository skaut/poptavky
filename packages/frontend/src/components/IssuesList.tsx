/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import type React from "react";
import ReactMarkdown from "react-markdown";
import { Link } from "react-router-dom";
import remarkGfm from "remark-gfm";

import type { Project } from "../interfaces/Project";
import type { ProjectInfo } from "../interfaces/ProjectInfo";
import type { ProjectIssue } from "../interfaces/ProjectIssue";
import type { ProjectListing } from "../interfaces/ProjectListing";
import { theme } from "../theme";
import { getIssueLink } from "../utils/getIssueLink";
import { H3 } from "./Typography";

export const IssuesList = ({
  issues,
  project,
}: {
  readonly issues: Array<ProjectIssue>;
  readonly project: ProjectListing | (Project & ProjectInfo);
}): React.JSX.Element => (
  <>
    {issues.map((issue) => (
      <article
        css={css`
          margin: 12px 0 24px;
          img {
            max-width: 100%;
          }
        `}
        key={issue.number}
      >
        <H3>
          <Link
            to={getIssueLink({
              project: {
                ...("info" in project
                  ? { ...project, ...project.info }
                  : { ...project }),
              },
              ...issue,
            })}
          >
            {issue.title}
          </Link>
          &nbsp;
          <span
            css={css`
              color: ${theme.colors.gray};
              font-weight: normal;
            `}
          >
            #{issue.number}
          </span>
        </H3>
        <ReactMarkdown remarkPlugins={[remarkGfm]}>
          {issue.description}
        </ReactMarkdown>
      </article>
    ))}
  </>
);
