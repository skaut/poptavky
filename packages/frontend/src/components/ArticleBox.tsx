import React from "react"
import ReactMarkdown from "react-markdown"
import styled from "@emotion/styled"
import { ColoredTag } from "./ColoredTag"
import { Link } from "react-router-dom"
import { Article, H2, Paragraph } from "./Typography"
import remarkGfm from "remark-gfm"

export const ArticleBox: React.FC<{
  title: string
  link?: string
  subtitle?: string
  subtitleLink?: string
  subtitleDescription?: string
  description: string
  tags?: string[]
}> = ({
  title,
  link,
  subtitle,
  subtitleLink,
  subtitleDescription,
  description,
  tags,
}) => (
  <ThinArticle>
    <H2>{link ? <Link to={link}>{title}</Link> : title}</H2>
    <ProjectName>
      {subtitle && subtitleLink && (
        <Link to={subtitleLink} title={subtitleDescription}>
          {subtitle}
        </Link>
      )}
    </ProjectName>
    <Paragraph>
      <ReactMarkdown remarkPlugins={[remarkGfm]}>{description}</ReactMarkdown>
    </Paragraph>
    {(tags ?? []).map((tag) => (
      <ColoredTag key={tag}>{tag}</ColoredTag>
    ))}
  </ThinArticle>
)

const ProjectName = styled("p")`
  margin: 0 0 0.6em;
  * {
    font-size: 0.8rem;
  }
`

const ThinArticle = styled(Article)`
  max-width: 600px;
`
