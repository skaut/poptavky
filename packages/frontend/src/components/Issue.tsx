import type React from "react";

import type { ProjectIssueWithProjectInfo } from "../utils/getAllIssues";
import { getIssueLink } from "../utils/getIssueLink";
import { getProjectLink } from "../utils/getProjectLink";
import { ArticleBox } from "./ArticleBox";

export const Issue = ({
  issue,
  hideTags = false,
}: {
  readonly issue: ProjectIssueWithProjectInfo;
  readonly hideTags?: boolean;
}): React.JSX.Element => (
  <ArticleBox
    description={issue.description}
    link={getIssueLink(issue)}
    subtitle={issue.project.name}
    subtitleDescription={issue.project["short-description"]}
    subtitleLink={getProjectLink(issue.project)}
    tags={!hideTags ? issue.project.tags : []}
    title={issue.title}
  />
);
