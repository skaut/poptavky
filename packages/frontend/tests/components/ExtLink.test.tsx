import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { ExtLink } from "../../src/components/ExtLink";

describe("ExtLink component", () => {
  test("should render correctly", () => {
    const { container } = render(
      <ExtLink href="http://www.somewhere.com">Link somewhere</ExtLink>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render correctly with title", () => {
    const { container } = render(
      <ExtLink href="http://www.somewhere.com" title="myTitle">
        Link somewhere
      </ExtLink>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render correctly with custom class", () => {
    const { container } = render(
      <ExtLink className="myClassName" href="http://www.somewhere.com">
        Link somewhere
      </ExtLink>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render correctly with target self", () => {
    const { container } = render(
      <ExtLink href="http://www.somewhere.com" targetSelf>
        Link somewhere
      </ExtLink>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
