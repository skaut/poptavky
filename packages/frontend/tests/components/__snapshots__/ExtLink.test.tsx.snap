// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExtLink component > should render correctly 1`] = `
<a
  class="css-ti9e4i"
  href="http://www.somewhere.com"
  rel="noreferrer noopener"
  target="_blank"
>
  Link somewhere
</a>
`;

exports[`ExtLink component > should render correctly with custom class 1`] = `
<a
  class="myClassName css-ti9e4i"
  href="http://www.somewhere.com"
  rel="noreferrer noopener"
  target="_blank"
>
  Link somewhere
</a>
`;

exports[`ExtLink component > should render correctly with target self 1`] = `
<a
  class="css-ti9e4i"
  href="http://www.somewhere.com"
>
  Link somewhere
</a>
`;

exports[`ExtLink component > should render correctly with title 1`] = `
<a
  class="css-ti9e4i"
  href="http://www.somewhere.com"
  rel="noreferrer noopener"
  target="_blank"
  title="myTitle"
>
  Link somewhere
</a>
`;
